openparticle/src.py

```diff
@@ -340,36 +340,81 @@ def to_list(self) -> List:
         return particle_op_list
 
     def parse(self) -> List:
+        """Merges neighboring creation/annihilation operators together into NumberOperators when acting on the same modes.
+
+        Merging only occurs when annihilation operators are immediately followed by creation operators. Therefore, there is
+            a preference for normal ordering the terms before calling this function.
+        """
```

Comment:
If you have $(a_0^\dagger)^3 (a_0)^2$, it won't add the first two creation operators to a stack, then collapse the inner $a_0^\dagger a_0$ to a `NumberOperator()` right?

Reply:
Nah it'll come out as `[BosonOperator(0, True), OccupationOperator("a", 0, power=2)]`. This is one of the test cases :D